src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -79,65 +87,82 @@ public virtual void FileWorkItems(string sarifLogFileContents)
 
         public virtual void FileWorkItems(SarifLog sarifLog)
         {
-            this.FilingSucceeded = false;
-            this.FiledWorkItems = new List<WorkItemModel>();
-
-            sarifLog = sarifLog ?? throw new ArgumentNullException(nameof(sarifLog));
-
-            this.FilingClient.Connect(this.FilingContext.PersonalAccessToken).Wait();
-
-            OptionallyEmittedData optionallyEmittedData = this.FilingContext.DataToRemove;
-            if (optionallyEmittedData != OptionallyEmittedData.None)
-            {
-                var dataRemovingVisitor = new RemoveOptionalDataVisitor(optionallyEmittedData);
-                dataRemovingVisitor.Visit(sarifLog);
-            }
-
-            optionallyEmittedData = this.FilingContext.DataToInsert;
-            if (optionallyEmittedData != OptionallyEmittedData.None)
+            using (IDisposable loggingScope = Logger.BeginScope("FileWorkItems"))
```

Comment:
>BeginScope [](start = 53, length = 10)

This is also optional. It's just an example, if anyone wants to use logging scopes.